lib/Doctrine/ORM/Query.php

```diff
@@ -322,9 +323,24 @@ protected function _doExecute()
 
         list($sqlParams, $types) = $this->processParameterMappings($paramMappings);
 
+        $this->evictResultSetCache($executor, $sqlParams, $types);
+
         return $executor->execute($this->_em->getConnection(), $sqlParams, $types);
     }
 
+    private function evictResultSetCache(AbstractSqlExecutor $executor, array $sqlParams, array $types)
+    {
+        if (null === $this->_queryCacheProfile || ! $this->getExpireResultCache()) {
+            return;
+        }
+
+        $cacheDriver = $this->_queryCacheProfile->getResultCacheDriver();
+
+        foreach ((array) $executor->getSqlStatements() as $statement) {
```

Comment:
It's indeed... cause: https://github.com/doctrine/doctrine2/pull/6417#discussion_r114082839